servlet-engine/src/main/java/io/micronaut/servlet/engine/DefaultServletHttpResponse.java

```diff
@@ -207,13 +239,17 @@ public void onComplete() {
                             }
                             flushIfReady();
                         }
-                        emitter.next(DefaultServletHttpResponse.this);
-                        emitter.complete();
+                        finish();
                     } catch (IOException e) {
                         emitter.error(e);
                     }
                 }
             }
+
+            private void finish() {
+                emitter.next(DefaultServletHttpResponse.this);
```

Comment:
@yawkat I had to add this after writing the error for it to work against core M7 otherwise it hung